src/main/java/bit/javaoop/FixedBudgetManager.java

```diff
@@ -0,0 +1,25 @@
+package bit.javaoop;
+
+public class FixedBudgetManager extends Manager {
+    private double budget;
+
+    public FixedBudgetManager(String name, double salary, double budget) {
+        super(name, salary);
+        this.budget = budget;
+    }
+
+    @Override
+    public boolean isSatisfied() {
+        return this.getSalary() > 20_000 && this.budget - super.myEmployeesSalaries() <= 15_000;
```

Comment:
Obliczanie ile nam jeszcze pieniędzy zostało w budżecie możnaby (ale nie trzeba) przenieść do osobnej metody - mogłoby być wtedy czytelniejsze. Zaznaczasz `this.budget - super.myEmployeesSalaries()` i klikasz ctrl + alt + m, nadajesz ładną nazwę, np. `getBudgetLeft()` albo `getActualBudget()` i ok.

"Magiczne stałe" możnaby gdzieś powyciągać do góry.
Czyli na przykład dodać pole w klasie Manager `protected final double managerSatisfyingSalary = 20_000`

Słówka kluczowego `this` raczej wolę unikać w takich metodkach. Intellij nam ładnie zmienia kolorek pól danej klasy, więc `this` nie trzeba używać. Raczej stosuje się go, gdy na prawdę na prawdę nie wiadomo skąd brana jest dana zmienna lub najczęściej po prostu w konstruktorach lub getterach, setterach, gdzie najczęściej pokrywają nam się nazwy zmiennych i trzeba je jakoś rozróżnić.